src/emc/usr_intf/stepconf/Submakefile

```diff
@@ -1,51 +1,69 @@
-STEPCONF_MODULES = pages build_INI build_HAL import_mach 
+STEPCONF_MODULES = pages axis build_INI build_HAL import_mach definitions preset
 
 STEPCONF_GLADEFILES := \
-	main_page.glade \
-	base.glade \
-	start.glade \
-	pport1.glade \
-	pport2.glade \
-	spindle.glade \
-	options.glade \
-	halui_page.glade \
-	axisx.glade \
-	axisy.glade \
-	axisz.glade \
-	axisu.glade \
-	axisv.glade \
-	axisa.glade \
-	finished.glade \
-
-PYTARGETS += ../bin/stepconf  ../lib/python/stepconf/__init__.py ../share/linuxcnc/linuxcnc-wizard.gif $(patsubst %,../lib/python/stepconf/%.py,$(STEPCONF_MODULES)) \
-	$(patsubst %, ../share/linuxcnc/stepconf/%, $(STEPCONF_GLADEFILES)) \
+	main_page \
+	base \
+	start \
+	pport1 \
+	pport2 \
+	spindle \
+	options \
+	gui_page \
+	halui_page \
+	axisx \
+	axisy \
+	axisz \
+	axisu \
+	axisv \
+	axisa \
+	finished
+
+#STEPCONF_PYFILES = $(patsubst %, emc/usr_intf/stepconf/%.py, $(STEPCONF_GLADEFILES))
+#STEPCONF_PYFILES := \
+#	gui_page
+
+STEPCONF_PYFILES = $(STEPCONF_GLADEFILES)
+
+PYTARGETS += ../bin/stepconf  ../lib/python/stepconf/__init__.py \
+	../share/linuxcnc/linuxcnc-wizard.gif $(patsubst %,../lib/python/stepconf/%.py,$(STEPCONF_MODULES)) \
+	$(patsubst %, ../share/linuxcnc/stepconf/%.glade, $(STEPCONF_GLADEFILES)) \
+	$(patsubst %, python_libs_for_galde_files/%.py, $(STEPCONF_PYFILES))
+	#$(patsubst %, emc/usr_intf/stepconf/%.py, $(STEPCONF_PYFILES))
 
 PYI18NSRCS += emc/usr_intf/stepconf/stepconf.py $(patsubst %,emc/usr_intf/stepconf/%.py,$(STEPCONF_MODULES)) \
-	$(patsubst %, emc/usr_intf/stepconf/%, $(STEPCONF_GLADEFILES)) \
+	$(patsubst %, emc/usr_intf/stepconf/%.glade, $(STEPCONF_GLADEFILES))
 
 ../lib/python/stepconf/__init__.py:
-	$(ECHO) building python init $(notdir $@)
+	$(ECHO) Stepconf: building python init $(notdir $@)
 	$(Q)mkdir -p ../lib/python/stepconf
 	$(Q)touch $@
 
 ../lib/python/stepconf/%.py: emc/usr_intf/stepconf/%.py
-	@$(ECHO) Syntax checking python script $(notdir $@)
+	@$(ECHO) Stepconf: syntax checking python libs $(notdir $@)
 	$(Q)$(PYTHON) -c 'import sys; compile(open(sys.argv[1]).read(), sys.argv[1], "exec")' $<
-	$(ECHO) Copying python script $(notdir $@)
+	$(ECHO) Stepconf: copying python libs $(notdir $@)
 	$(Q)mkdir -p ../lib/python/stepconf
 	$(Q)cp $< $@
 
 ../share/linuxcnc/stepconf/%.glade: emc/usr_intf/stepconf/%.glade
-	$(ECHO) Copying glade file $(notdir $@)
+	$(ECHO) Stepconf: copying glade file $(notdir $@)
 	$(Q)mkdir -p $(dir $@)
 	$(Q)cp -f $< $@
 
+python_libs_for_galde_files/%.py:
```

Comment:
I suggest a change similar to this, but I will apply it myself after applying your changes if that makes things easier on you.

~~~~
From fd75ad163488574ff69f894f7a60fcb4ce1cdbde Mon Sep 17 00:00:00 2001
From: Jeff Epler <jepler@unpythonic.net>
Date: Wed, 26 Jul 2017 21:07:20 -0500
Subject: [PATCH] stepconf: fix Makefile rules so they only run when they need
 to

.. before, every time you ran "make", it would operate on these files
even if they were unchanged.

There was also a typo in a variable name in the Submakefile, but that
variable is now removed.
---
 src/emc/usr_intf/stepconf/Submakefile | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/src/emc/usr_intf/stepconf/Submakefile b/src/emc/usr_intf/stepconf/Submakefile
index 26a80514b..d7a889643 100644
--- a/src/emc/usr_intf/stepconf/Submakefile
+++ b/src/emc/usr_intf/stepconf/Submakefile
@@ -27,8 +27,7 @@ STEPCONF_PYFILES = $(STEPCONF_GLADEFILES)
 PYTARGETS += ../bin/stepconf  ../lib/python/stepconf/__init__.py \
 	../share/linuxcnc/linuxcnc-wizard.gif $(patsubst %,../lib/python/stepconf/%.py,$(STEPCONF_MODULES)) \
 	$(patsubst %, ../share/linuxcnc/stepconf/%.glade, $(STEPCONF_GLADEFILES)) \
-	$(patsubst %, python_libs_for_galde_files/%.py, $(STEPCONF_PYFILES))
-	#$(patsubst %, emc/usr_intf/stepconf/%.py, $(STEPCONF_PYFILES))
+	$(patsubst %, ../lib/python/stepconf/%.py, $(STEPCONF_PYFILES))
 
 PYI18NSRCS += emc/usr_intf/stepconf/stepconf.py $(patsubst %,emc/usr_intf/stepconf/%.py,$(STEPCONF_MODULES)) \
 	$(patsubst %, emc/usr_intf/stepconf/%.glade, $(STEPCONF_GLADEFILES))
@@ -50,12 +49,12 @@ PYI18NSRCS += emc/usr_intf/stepconf/stepconf.py $(patsubst %,emc/usr_intf/stepco
 	$(Q)mkdir -p $(dir $@)
 	$(Q)cp -f $< $@
 
-python_libs_for_galde_files/%.py:
-	$(ECHO) Stepconf: syntax checking gui python libs emc/usr_intf/stepconf/$(notdir $@)
-	$(Q)$(PYTHON) -c 'import sys; compile(open(sys.argv[1]).read(), sys.argv[1], "exec")' emc/usr_intf/stepconf/$(notdir $@)
-	$(ECHO) Stepconf: copying python gui libs emc/usr_intf/stepconf/$(notdir $@)
+$(patsubst %, ../lib/python/stepconf/%.py, $(STEPCONF_PYFILES)): ../lib/python/stepconf/%.py: emc/usr_intf/stepconf/%.py
+	$(ECHO) Stepconf: syntax checking gui python libs $*
+	$(Q)$(PYTHON) -c 'import sys; compile(open(sys.argv[1]).read(), sys.argv[1], "exec")' $<
+	$(ECHO) Stepconf: copying python gui libs $*
 	$(Q)mkdir -p ../lib/python/stepconf
-	$(Q)cp emc/usr_intf/stepconf/$(notdir $@) ../lib/python/stepconf/$(notdir $@)
+	$(Q)cp -f $< $@
 
 ../share/linuxcnc/linuxcnc-wizard.gif: ../linuxcnc-wizard.gif
 	$(ECHO) Stepconf: copying $(notdir $@)
-- 
2.11.0
~~~~